ruby/templates/api_client_typhoeus_partial.mustache

```diff
@@ -1,17 +1,24 @@
+    require 'securerandom'
+    
     # Call an API with given options.
     #
     # @return [Array<(Object, Integer, Hash)>] an array of 3 elements:
     #   the data deserialized from response body (could be nil), response status code and response headers.
     def call_api(http_method, path, opts = {})
+      opts[:'header_params'][:'svix-req-id'] = SecureRandom.random_number( 2**32 ).to_s
       request = build_request(http_method, path, opts)
       response = request.run
 
       # retry 500s
       sleep_time = 0.05
+      retry_count = 0
       (0..1).each do |n|
-        break if response.success? && response.code < 500
+        break if response.success? || response.code < 500
```

Comment:
You changed behavior here, on purpose? And why?

Reply:
Yeah it wasn't working as intended before, it was retrying on 4xx's as well

Reply:
Ah yeah.